src/main/java/com/minecolonies/coremod/colony/managers/StatisticAchievementManager.java

```diff
@@ -0,0 +1,273 @@
+package com.minecolonies.coremod.colony.managers;
+
+import com.minecolonies.coremod.achievements.MineColoniesAchievement;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.TriggerColonyAchievements;
+import com.minecolonies.coremod.util.AchievementUtils;
+import net.minecraft.advancements.Advancement;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.ColonyConstants.*;
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+public class StatisticAchievementManager implements IStatisticAchievementManager
+{
+    /**
+     * Statistical values.
+     */
+    private int minedOres         = 0;
+    private int minedDiamonds     = 0;
+    private int harvestedWheat    = 0;
+    private int harvestedPotatoes = 0;
+    private int harvestedCarrots  = 0;
+    private int killedMobs        = 0;
+    private int builtHuts         = 0;
+    private int caughtFish        = 0;
+    private int felledTrees       = 0;
+    private int plantedSaplings   = 0;
+
+    /**
+     * List of achievements within the colony.
+     */
+    @NotNull
+    private final List<Advancement> colonyAchievements = new ArrayList<>();
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound, @NotNull final Colony colony)
+    {
+        // Restore colony achievements
+        final NBTTagList achievementTagList = compound.getTagList(TAG_ACHIEVEMENT_LIST, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < achievementTagList.tagCount(); ++i)
+        {
+            final NBTTagCompound achievementCompound = achievementTagList.getCompoundTagAt(i);
+            final String achievementKey = achievementCompound.getString(TAG_ACHIEVEMENT);
+
+            //todo serialization
+            /*final StatBase statBase = StatList.getOneShotStat(achievementKey);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)